api/goodreads.py

```diff
@@ -0,0 +1,42 @@
+from requests import get
+from configuration import config
+import xml.etree.ElementTree as ET
+import isbnlib
+
+
+def goodreads(update, context):
+    """ Command to query GoodReads for a book"""
+    message = update.message
+    query = ' '.join(context.args)
+
+    if not query:
+        text = "*Usage:* `/gr {BOOK_NAME}`\n"\
+               "*Example:* `/gr stoner`"
+    else:
+        response = get(f'https://www.goodreads.com/search.xml?key={config["GOODREADS_API_KEY"]}&q={query}')
+        root = ET.fromstring(response.content)
+
+        for node in root.iter('work'):
```

Comment:
@Tyagdit: the query response is XML with the actual results being under `results -> work`. The `results` field is an array of `work` objects which are each search results. Right now as soon as we find the first, we break out of the for loop. 

Here is an example query response from Goodreads: https://www.goodreads.com/search/index.xml?key=qyYkqnJ6FWgHzZaG9py5g&q=Ender%27s+Game&search=isbn

Any way to return the list of results as an inline list, similar to how @imdb (on Telegram) does it?

Reply:
To do that we will need an `InlineQueryHandler` instead of `CommandHandler`. In the callback for that we need to make a list of `InlineQueryResultArticle` with the title and such which will show up in the list above the text box. In each of those we need a `InputTextMessageContent` which will be the message sent when the user selects that entry.  We can paginate this list as well. [Here's an example](https://github.com/python-telegram-bot/python-telegram-bot/blob/master/examples/inlinebot.py) for reference

I suggest we use the result list we have atm to show the thumbnail, name, author, rating and year in the inline list and then use [this endpoint](https://www.goodreads.com/api/index#book.show) to get the description, ISBN, series, characters, thumbnail and link to the goodreads page. We'll have to make sure that we request these details only when the user selects an entry.